### PR TITLE
Added filtering skipped process functionality

### PR DESCRIPTION
**Pull request by manasa123104**
I have implemented an enhancement to the filtering functionality within the history table. Specifically, I have introduced the inclusion of "Skipped" within the dropdown menu of available filtering processes. This enhancement aims to provide users with a comprehensive set of options for filtering processes. By adding "Skipped" to the filtering options, users can now conveniently filter and analyze data based on processes that have been intentionally omitted or bypassed, enriching their analytical capabilities and improving overall workflow efficiency.

### CODE REVIEW

File: src/main/java/com/gw/tools/ProcessTool.java [hunk 1]
```diff
@@ -737,7 +737,7 @@ public String all_history(String pid, boolean ignore_skipped) {
 	
 	public String all_history(String pid) {
 
-		return all_history(pid, true);
```
Comment (gokulprathin8) [inline src/main/java/com/gw/tools/ProcessTool.java:740]: this should be dynamic. Passing true always will show skipped process always.
As of now it is hardcoded to be false. So lets make sure we only show skipped when we click a button or use a certain filter and not show it always.

File: src/main/java/com/gw/tools/ProcessTool.java [hunk 2]
```diff
@@ -737,7 +737,7 @@ public String all_history(String pid, boolean ignore_skipped) {
...
+		return all_history(pid, false);
```
Comment (gokulprathin8) [inline src/main/java/com/gw/tools/ProcessTool.java:740]: As of now we are not explicity showing skipped status in the history. I beleive it is set that way for a reason. This change will make it show always. I think only when the filter is active we should be showing skipped.

**Review by gokulprathin8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/gw/tools/ProcessTool.java (hunk 2 above)
Comment (gokulprathin8) [reply src/main/java/com/gw/tools/ProcessTool.java:740]: @ZihengSun might have a better understanding for hiding the skipped processes by default
skipped processes might make it look cluttered is my take on this and we should not be showing it by default.

Comment (manasa123104) [reply src/main/java/com/gw/tools/ProcessTool.java:740]: I have made the required changes. Please check.

Comment (manasa123104) [reply src/main/java/com/gw/tools/ProcessTool.java:740]: Changes can be viewed in [Extended the functionality of the filtering processes](https://github.com/ESIPFed/Geoweaver/pull/475/commits/5ba619a62de6331f50e84d78935c46679138c9a8)

File: src/main/resources/static/js/gw.history.js [hunk 3]
```diff
@@ -287,6 +288,13 @@ GW.history = {
 
         content += "</tbody>";
 
+        $('#statusFilter').on('change', function () {
+            alert("ok");
+            console.log("status filter changed");
```
Comment (gokulprathin8) [inline src/main/resources/static/js/gw.history.js:293]: is alert needed here?

File: src/main/resources/static/js/gw.history.js [hunk 4]
```diff
@@ -427,21 +441,37 @@ GW.history = {
...
-            function(settings, data, dataIndex) {
+        $(document).ready(function() {
+            // Function to apply search filter
+            $.fn.dataTable.ext.search.push(function(settings, data, dataIndex) {
                 const selectedStatus = $('#statusFilter').val().toLowerCase();
                 const rowStatus = data[4].toLowerCase();
-
+        
+                console.log("Selected Status:", selectedStatus);
+                console.log("Row Status:", rowStatus);
```
Comment (gokulprathin8) [inline src/main/resources/static/js/gw.history.js:451]: can you clean this up? this might print like 50 rows each time to console.

**Review by gokulprathin8 — COMMENTED** [review]
Functioanlity works fine!
Just a few minor cleanup. Can merge after the cleanup
Good work @manasa123104 !

**Comment by gokulprathin8** [comment]
the changes seem fine. 
the formatting of the file makes it hard to read pin point the changes done in the PR though.

**Comment by gokulprathin8** [comment]
works fine. please take of formatting from next time @meghanakr7

**Review by gokulprathin8 — APPROVED** [review]

**Pull request merged**